app/javascript/components/widget-controller.jsx

```diff
@@ -33,234 +29,58 @@ const getWidgets = gql`
+      variables: {
+        id: nextWidgetId,
+        cityName,
+      },
     });
   };
 
-  handleClicks = ({ target }) => {
-    const targetId = target.id;
-    if (targetId.startsWith('widget_')) {
-      const widgetId = parseInt(targetId.replace('widget_', ''), 10);
-      this.switchToPage(widgetId);
-    }
-  };
-
-  render() {
-    const { currentWidgetId, pausedTime } = this.state;
-
-    const fetchPolicy = currentWidgetId === 0 ? 'network-only' : 'cache-first';
-
-    const { cityName } = this.props;
-
-    return (
-      <Query
-        query={getWidgets}
-        variables={{ id: currentWidgetId, cityName }}
-        fetchPolicy={fetchPolicy}
-        onCompleted={response => {
-          const { location } = response;
-          const { widgets } = location;
-          const { enabled: enabledWidgets, byIdOrFirst: current } = widgets;
-
-          const index = enabledWidgets.findIndex(w => w.id === current.id);
-
-          const prevWidgetId =
-            enabledWidgets[mathMod(index - 1, enabledWidgets.length)].id;
-          const nextWidgetId =
-            enabledWidgets[mathMod(index + 1, enabledWidgets.length)].id;
-
-          this.setState({
-            currentWidgetId: current.id,
-            nextWidgetId,
-            prevWidgetId,
-            transitionTime: Date.now() + current.durationSeconds * 1000,
-          });
-        }}
-      >
-        {({ loading, error, data }) => {
-          if (loading) {
-            return (
-              <Wrapper>
-                <FullPanel>
-                  <LoadingMessage />
-                </FullPanel>
-                <SidePanel
-                  widgets={[]}
-                  selectedWidgetId={0}
-                  showWeather={false}
-                  totalTime={0}
-                  isPaused={false}
-                  cityName={cityName}
-                />
-              </Wrapper>
-            );
-          }
-
-          if (error) {
-            // eslint-disable-next-line
-            console.error(error);
-            return (
-              <Wrapper>
-                <FullPanel>
-                  <DisconnectedMessage />
-                </FullPanel>
-                <SidePanel
-                  widgets={[]}
-                  selectedWidgetId={0}
-                  showWeather={false}
-                  totalTime={0}
-                  isPaused={false}
-                  cityName={cityName}
-                />
-              </Wrapper>
-            );
-          }
-
-          const { location } = data;
-          const { widgets } = location;
-          const { enabled: enabledWidgets, byIdOrFirst: current } = widgets;
+  return (
+    <Query
+      query={getWidgets}
+      variables={{ id: newWidgetId, cityName }}
+      fetchPolicy={fetchPolicy}
+      onCompleted={response => {
+        const {
+          location: {
+            widgets: { byIdOrFirst: current },
+          },
+        } = response;
+        setNewWidgetId(current.id);
```

Comment:
I'm still confused by this I think. If the API returns the `current.id`, why do we need to keep it in state?